### PR TITLE
fix: Refine compactionTask to avoid data race

### PR DESCRIPTION
**Pull request by wayblink**
issue: #36897

### CODE REVIEW

**Review by tedxu — COMMENTED** [review]
/lgtm

It duplicates #36965. Will close that one if this one is checked in.

File: internal/datacoord/compaction_task_clustering.go [hunk 1]
```diff
@@ -47,9 +48,9 @@ import (
...
 
 type clusteringCompactionTask struct {
-	*datapb.CompactionTask
-	plan   *datapb.CompactionPlan
-	result *datapb.CompactionPlanResult
+	taskProto atomic.Value // *datapb.CompactionTask
+	plan      *datapb.CompactionPlan
+	result    *datapb.CompactionPlanResult
 
 	span             trace.Span
```
[the remark this answers is not in the document]
Comment (wayblink) [reply internal/datacoord/compaction_task_clustering.go:55]: It seems the span is not currently in use and can be safely removed without any issues.

[the remark this answers is not in the document]
Comment (wayblink) [reply internal/datacoord/compaction_task_clustering.go:55]: we can remove it or complete the logic later.

**Pull request merged**